src/rgw/rgw_asio_frontend.cc

```diff
@@ -792,11 +804,13 @@ void AsioFrontend::accept(Listener& l, boost::system::error_code ec)
       [this, s=std::move(socket)] (spawn::yield_context yield) mutable {
         Connection conn{s};
         auto c = connections.add(conn);
-        // wrap the socket in an ssl stream
-        ssl::stream<tcp::socket&> stream{s, *ssl_context};
+        boost::beast::tcp_stream timeout_stream((std::move(s)));
```

Comment:
could simplify this a bit if you construct the `tcp_stream` outside of the lambda:
```diff
   }
-  auto socket = std::move(l.socket);
+  boost::beast::tcp_stream stream(std::move(l.socket));
   tcp::no_delay options(l.use_nodelay);
```
and move it into each lambda with `s=std::move(stream)`